# strelkasomatic: Output

## Introduction

The directories listed below will be created in the results directory after the pipeline has finished. All paths are relative to the top-level results directory.

<!-- TODO nf-core: Write this documentation describing your workflow's output -->

# Pipeline output overview

The pipeline is built using [Nextflow](https://www.nextflow.io/) and processes data using the following steps:

- [manta_out](#manta_out) - call candidate indels and SVs
- [strelka_out](#strelka_out) - call the final SNVs and Indels
- [Pipeline information](#pipeline-information) - Report metrics generated during the workflow execution

### manta_out

<details markdown="1">
<summary>Output files</summary>

- `manta_out/`
  - `${sample_id}/results/`
    - `evidence/`: part of manta output
    - `stats/`: part of manta output
      - `alignmentStatsSummary.txt`
      - `svCandidateGenerationStats.tsv`
      - `svCandidateGenerationStats.xml`
      - `svLocusGraphStats.tsv`
    - `variants/`: part of manta output
      - `candidateSmallIndels.vcf.gz`
      - `candidateSmallIndels.vcf.gz.tbi`
      - `candidateSV.vcf.gz`
      - `candidateSV.vcf.gz.tbi`
      - `diploidSV.vcf.gz`
      - `diploidSV.vcf.gz.tbi`
      - `somaticSV.vcf.gz`
      - `somaticSV.vcf.gz.tbi`

</details>

Details regarding output information are found [here](https://github.com/Illumina/manta/tree/master/docs/userGuide#outputs)

[Manta](https://academic.oup.com/bioinformatics/article/32/8/1220/1743909?login=true) a method to discover structural variants and indels from next generation sequencing data

### strelka_out

<details markdown="1">
<summary>Output files</summary>

- `strelka_out/results/`
  - `${sample_id}/results/`
    - `stats/`: part of strelka output
      - `runStats.tsv`
      - `runStats.xml`
    - `variants/`: part of strelka output
      - `somatics.indels.vcf.gz`
      - `somatics.indels.vcf.gz.tbi`
      - `somatics.snvs.vcf.gz`
      - `somatics.snvs.vcf.gz.tbi`

</details>

Details regarding output information are found [here](https://github.com/Illumina/strelka/tree/v2.9.x/docs/userGuide#outputs)

[Strelka2](https://www.nature.com/articles/s41592-018-0051-x) an open-source small-variant-calling method for research and clinical germline and somatic sequencing applications


### Pipeline information

<details markdown="1">
<summary>Output files</summary>

- `pipeline_info/`
  - Reports generated by Nextflow: `execution_report.html`, `execution_timeline.html`, `execution_trace.txt` and `pipeline_dag.dot`/`pipeline_dag.svg`.
  - Reports generated by the pipeline: `pipeline_report.html`, `pipeline_report.txt` and `software_versions.yml`. The `pipeline_report*` files will only be present if the `--email` / `--email_on_fail` parameter's are used when running the pipeline.
  - Reformatted samplesheet files used as input to the pipeline: `samplesheet.valid.csv`.
  - Parameters used by the pipeline run: `params.json`.

</details>

[Nextflow](https://www.nextflow.io/docs/latest/tracing.html) provides excellent functionality for generating various reports relevant to the running and execution of the pipeline. This will allow you to troubleshoot errors with the running of the pipeline, and also provide you with other information such as launch commands, run times and resource usage.
